### PR TITLE
Ak96/automate contributer list

### PR DESCRIPTION
**Pull request by aaruni96**
This script will eventually be used to automate the updating of the contributer's list on the website.

cc @HereAround

### CODE REVIEW

**Comment by HereAround** [comment]
Some minor text change to the list:
* Remove "as of August 2024".
* Add footnote "This site is updated on the first of every month" (if that is what we decide the cronjob is doing).

**Comment by aaruni96** [comment]
This is now in a pretty good state now. The API_KEY needs to be populated server side, which I don't have access to right now.

A summary of changes :

 - split `contributors.md` to `people_list.yml` and `contributors.md` : the yaml file has the actual data, and the md file just controls how it is shown
 - python script writes updated people list to `autogen_people_list.yml`. This is gitignored, and generated by the python script. python script added to `update.sh`
 - website actually reads data from this autogenerated yaml file, and will show 0 contributors if the file is not found.

cc @HereAround . We can ask max to look at it once we both agree there's no more changes to be done (plus the CI is a bit more complicated than I thought, so we will need help with the cron as well)

**Comment by HereAround** [comment]
I think the current state is ready for discussion. @fingolfin could you please take a look. Thank you!

File: _data/people_list.yml [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+  github: ederc
+  status: active
+
+- name: Nicolas Faross
+  github: pinguly
+  affiliation: Saarland University
+  email: faross@math.uni-sb.de
+  website: https://www.uni-saarland.de/lehrstuhl/weber-moritz/team/nicolas-faross.html
+  status: active
+  comment: who ?
```
Comment (lgoettgens) [inline _data/people_list.yml]: PhD student of Moritz Weber in Saarbrücken

File: _data/people_list.yml [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+  email: lukas.kuehne@math.uni-bielefeld.de
+  github: LukasKuehne
+  status: active
+
+- name: Markus Kurtz
+  affiliation: University of Kaiserslautern-Landau
+  email: kurtz@mathematik.uni-kl.de
+  github: mgkurtz
+  status: active
+  comment: ask Max?
```
Comment (lgoettgens) [inline _data/people_list.yml]: Iirc former student assistant in KL

File: _data/people_list.yml [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+- name: Zoe Geiselmann
+  affiliation: TU Berlin
+  website: https://www.tu.berlin/dmg/ueber-uns/team
+  email: geiselmann@math.tu-berlin.de
+  github: zkgeiselmann
+  status: active
+
+- name: Lars Göttgens
+  affiliation: RWTH Aachen University
+  website: https://www.art.rwth-aachen.de/cms/MATHB/Der-Lehrstuhl/Team/Wissenschaftliche-Beschaeftigte/~bbxbwx/Lars-Goettgens/
```
Comment (lgoettgens) [inline _data/people_list.yml]:
```suggestion
  website: https://www.art.rwth-aachen.de/cms/~bbxbwx/
```

File: _data/people_list.yml [hunk 4]
```diff
@@ -0,0 +1,618 @@
...
+- name: Morgan Rodgers
+  affiliation: University of Kaiserslautern-Landau
+  website: https://math.rptu.de/ags/agag/personen/mitglieder
+  email: morgan.rodgers@rptu.de
+  github: mjrodgers
+  status: active
+
+- name: Felix Röhrich
+  affiliation: RWTH Aachen University
+  website: https://www.art.rwth-aachen.de/cms/mathb/der-lehrstuhl/team/~rekt/mitarbeiter-campus-/?gguid=0x2C0E76EAF4EC9645BA4B35EE89317E58&lidx=1&allou=1
```
Comment (lgoettgens) [inline _data/people_list.yml]:
```suggestion
  website: https://www.art.rwth-aachen.de/cms/~xlgua/
```

File: _data/people_list.yml [hunk 5]
```diff
@@ -0,0 +1,618 @@
...
+- name: Kamillo Ferry
+  affiliation: TU Berlin
+  website: https://kafe.dev/
+  email: kafe@kafe.dev
+  github: ooinaruhugh
+  status: active
+
+- name: Ghislain Fourier
+  affiliation: RWTH Aachen University
+  website: https://www.art.rwth-aachen.de/cms/mathb/der-lehrstuhl/team/professorinnen-und-professoren/~rnko/fourier/?allou=1
```
Comment (lgoettgens) [inline _data/people_list.yml]:
```suggestion
  website: https://www.art.rwth-aachen.de/cms/~rnko/
```
More persistent url (same for other people of our chair below)

File: etc/update-contributors.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+# but we also need to include the autogen file if available
+# to keep a better track of retired people
+if os.path.isfile("../_data/autogen_people_list.yml"):
+    infile = "../_data/autogen_people_list.yml"
+else:
+    infile = "../_data/people_list.yml"
+with open(infile, "r") as ymlfile:
+    peopleList = yaml.safe_load(ymlfile)
+names = [i['github'] for i in peopleList]
+repoList = ["thofma/hecke.jl", "oscar-system/Oscar.jl", "Nemocas/Nemo.jl",
```
Comment (lgoettgens) [inline etc/update-contributors.py]:
```suggestion
repoList = ["thofma/Hecke.jl", "oscar-system/Oscar.jl", "Nemocas/Nemo.jl",
```

File: _data/people_list.yml (hunk 2 above)
Comment (fingolfin) [reply _data/people_list.yml]: correct

Comment (fingolfin) [reply _data/people_list.yml]: he did his masters with me

File: _data/people_list.yml [hunk 7]
```diff
@@ -0,0 +1,618 @@
...
+  comment: Associated with Anne Fruhbig Kruger as of 2024
+
+- name: Sreevatsa Sasisekar
+  github: Sreevatsa2k
+  email: sasiseka@mathematik.uni-kl.de
+  affiliation: University of Kaiserslautern-Landau
+  status: active
+  comment: Master student of Claus Fieker as of 2024
+
+- name: Bjoern Schaefer
```
Comment (fingolfin) [inline _data/people_list.yml]:
```suggestion
- name: Björn Schäfer
```

**Comment by fingolfin** [comment]
I like it. As we just discussed, I think we could set up a GH Action that runs periodically and uses https://github.com/peter-evans/create-pull-request to open a pull request which updates the contributors YAML file in the repository.

This means one extra step to "activate" the changes. But I think that's a good thing, plus we see in the repo what is on the website.

File: _data/people_list.yml (hunk 1 above)
Comment (aaruni96) [inline _data/people_list.yml]:
```suggestion
  comment: PhD student of Moritz Weber in Saarbruecken, as of 2024
```

File: _data/people_list.yml (hunk 2 above)
Comment (aaruni96) [inline _data/people_list.yml]:
```suggestion
  comment: Former master student of Max Horn
```

File: _data/people_list.yml [hunk 8]
```diff
@@ -0,0 +1,618 @@
...
+  affiliation: University of Kaiserslautern-Landau
+  status: retired
+  github: tthsqe12
+  comment: who ?
+
+- name: Carlo Sircana
+  affiliation: University of Kaiserslautern-Landau
+  status: retired
+  comment: PhD student of Claus Fieker
+  github: CarloSircana
```
Comment (fingolfin) [inline _data/people_list.yml]: Will your script also put these comments in a fixed order?

And sort the entries in this file overall?

File: _data/people_list.yml [hunk 9]
```diff
@@ -0,0 +1,618 @@
...
+  email: sachinkmohan@proton.me
+  github: sachinkm308
+  status: retired
+  comment: Who ?
+
+- name: Oleksandr Motsak
+  affiliation: IMAGINARY
+  status: retired
+  github: malex984
+  comment: Who ?
```
Comment (fingolfin) [inline _data/people_list.yml]: Former student (phd?) in Kaiserslautern (@fieker or @wdecker can clarify)

File: _data/people_list.yml [hunk 10]
```diff
@@ -0,0 +1,618 @@
...
+  status: retired
+  comment: Who ?
+
+- name: Frank Lübeck
+  affiliation: RWTH Aachen University
+  website: http://www.math.rwth-aachen.de/~Frank.Luebeck/index.html
+  email: Frank.Luebeck@math.rwth-aachen.de
+  github: frankluebeck
+  status: retired
+
```
Comment (fingolfin) [inline _data/people_list.yml]: I am not aware of any direct contribution by Frank to OSCAR, so perhaps he should be removed here.
```suggestion
- name: Frank Lübeck
  affiliation: RWTH Aachen University
  website: http://www.math.rwth-aachen.de/~Frank.Luebeck/index.html
  email: Frank.Luebeck@math.rwth-aachen.de
  github: frankluebeck
  status: retired
```

File: _data/people_list.yml [hunk 11]
```diff
@@ -0,0 +1,618 @@
...
+  status: retired
+  comment: Freelance mathematician, formerly at KIT
+
+- name: Avinash (Avi) Kulkarni
+  affiliation: Dartmouth College
+  website: https://math.dartmouth.edu/~akulkarn/index.html
+  email: avinash.a.kulkarni@dartmouth.edu
+  github: a-kulkarn
+  status: retired
+  comment: Who ?
```
Comment (fingolfin) [inline _data/people_list.yml]: @fieker can fill this in

File: _data/people_list.yml [hunk 12]
```diff
@@ -0,0 +1,618 @@
...
+  affiliation: University of Western Ontario
+  website: https://sites.google.com/view/taylorbrysiewicz/home
+  email: tbrysiew@uwo.ca
+  github: tbrysiewicz
+  status: retired
+
+- name: Giovanni De Franceschi
+  affiliation: University of Kaiserslautern-Landau
+  status: retired
+  comment: Group theory - who ?
```
Comment (fingolfin) [inline _data/people_list.yml]: did a postdoc in KL around 2021 (plus minus, would have to look it up, but probably good enough).

BTW I think it's a great idea you started to add years to some of this info

File: _data/people_list.yml (hunk 8 above)
Comment (HereAround) [reply _data/people_list.yml]: * The fields in each yml entry are sorted alphabetically. So "comment" before "github".
* The yml entries themselves are sorted (by the python script) according to last name of contributor.

File: _data/people_list.yml (hunk 12 above)
Comment (HereAround) [inline _data/people_list.yml]:
```suggestion
  comment: Was PostDoc in Kaiserslautern around 2021.
```

File: _data/people_list.yml [hunk 13]
```diff
@@ -0,0 +1,618 @@
...
+  github: a-kulkarn
+  status: retired
+  comment: Who ?
+
+- name: Frank Lübeck
+  affiliation: RWTH Aachen University
+  website: http://www.math.rwth-aachen.de/~Frank.Luebeck/index.html
+  email: Frank.Luebeck@math.rwth-aachen.de
+  github: frankluebeck
+  status: retired
```
Comment (HereAround) [inline _data/people_list.yml]:
```suggestion
```

Comment (HereAround) [reply _data/people_list.yml]: As suggested by @fingolfin .

**Comment by HereAround** [comment]
Let us add a comment where people can add their data (no longer in .md and so cannot click "edit this page" anymore) - as suggested by @fingolfin .

File: _data/people_list.yml [hunk 14]
```diff
@@ -0,0 +1,618 @@
...
+  github: gvanuxem
+  status: active
+  comment: Who ?
+
+- name: Laura Voggesberger
+  affiliation: Ruhr-Universität Bochum
+  website: https://www.ruhr-uni-bochum.de/ffm/Lehrstuehle/Lehrstuhl-VI/voggesberger.html
+  email: laura.voggesberger@ruhr-uni-bochum.de
+  github: voggesbe
+  status: active
```
Comment (joschmitt) [inline _data/people_list.yml]:
```suggestion
- name: Laura Voggesberger
  affiliation: Ruhr-Universität Bochum
  github: voggesbe
  status: active
```
Laura is not in Bochum anymore and left academia.

File: _data/people_list.yml [hunk 15]
```diff
@@ -0,0 +1,618 @@
...
+  email: alexander.dinges@cs.rptu.de
+  github: AlexD97
+  status: retired
+  comment: Worked for Janko on modules
+
+- name: Raul Epure
+  affiliation: University of Kaiserslautern-Landau
+  status: retired
+  github: raulepure
+  comment: Used to work in TU KL, under whom and for what ? who ?
```
Comment (joschmitt) [inline _data/people_list.yml]: He was a PhD student of Mathias Schulze until ~2021.

**Comment by aaruni96** [comment]
Merging as @HereAround  and I think its okay to merge this and see the CI in action

**Pull request merged**